### DISCUSSION TITLE
I want to improve Z accuracy of a v400

### DISCUSSION THREAD

**Discussion by Yog-Sothoth68** [Guilouz/Klipper-Flsun-Speeder-Pad#174] [Q&A]
Hi all,

I have a persistent problem with my v400. I've had it for several months now (4th printer) and it's really amazing. However I have a problem with dimensional accuracy. Indeed, the height (Z axis) of my printed parts is much higher than that of the model (displayed) in the slicer.

For example on an XYZ cube, my faces X and Y are rather precise (X = 20.01 and Y = 19.99) but in Z I am almost at 20.30.

I recently installed the Guilouz firmware (I thank him in passing, the work is incredible), thinking that my problem would disappear (like magic ^^). I spent several hours recalibrating my machine, first those recommended on this page: https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Calibrate-your-Printer, then an advanced calibration (https://www.klipper3d.org/Delta_Calibrate.html), I redid a Bed mesh, I recalibrated the flow then the pressure advance.

Finally I relaunched an XYZ cube and surprise surprise.... the result is the same, always a height in Z almost 102% of normal.

What calibration can I make to improve the dimensional accuracy of my v400. On any other of my machines I modify the rotation distance of my steppers but the only time I tried on a v400 I ruined my bed....

I've searched everywhere (here, on you tube, other forums), but I seem to be the only one having this problem. If other people have already posted this type of question, I am sorry for reposting and ask you to give me the link to their post including an answer.

I would like to make technical parts on this machine that need to fit together and I admit that this defect is quite disturbing.

Thank you in advance for your help!

**Comment by brdi** [comment]
Hello YS,
I've done the Guilouz/Klipper update on my V400 and sofar never experienced problems when printing PLA.
I did the calibration according to https://github.com/eeichhorn/Flsun-v400/tree/main/calibration.
I am now printing a lot of stuff in PETG. Harder to get speedy prints and I experience some issues on the dimensions. 
Not specifically onthe Z-Axe but then again I have not really checked the output. Will do that and get back to you if i find something worth mentioning.
